In [1]:
import pandas as pd
from __future__ import division

Location = "titanic3.xls"
df = pd.read_excel(Location)

In [5]:
df.head()

pclass  survived                                             name     sex  \
0       1         1                    Allen, Miss. Elisabeth Walton  female   
1       1         1                   Allison, Master. Hudson Trevor    male   
2       1         0                     Allison, Miss. Helen Loraine  female   
3       1         0             Allison, Mr. Hudson Joshua Creighton    male   
4       1         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   

       age  sibsp  parch  ticket      fare    cabin embarked boat   body  \
0  29.0000      0      0   24160  211.3375       B5        S    2    NaN   
1   0.9167      1      2  113781  151.5500  C22 C26        S   11    NaN   
2   2.0000      1      2  113781  151.5500  C22 C26        S  NaN    NaN   
3  30.0000      1      2  113781  151.5500  C22 C26        S  NaN  135.0   
4  25.0000      1      2  113781  151.5500  C22 C26        S  NaN    NaN   

                         home.dest  
0                     St Louis, MO  
1  Montreal, PQ / Chesterville, ON  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON

***Probability of survival for female passengers from the first class, 60 and more years old. 

In [6]:
df_survived = df[df['survived']==1]
df_women = df[df['sex']=='female']
df_wealthy = df[df['pclass']==1]
df_old = df[df['age']>=60]

In [11]:
p_lived=len(df_survived)/len(df)
p_lived

0.3819709702062643

In [13]:
df_survivedwealthy=df_survived[df_survived['pclass']==1]

In [15]:
p_wealthy_survived=len(df_survivedwealthy)/len(df_survived)
p_wealthy_survived

0.4

In [16]:
df_survivedwomen=df_survived[df_survived['sex']=='female']

In [17]:
p_women_survived=len(df_survivedwomen)/len(df_survived)
p_women_survived

0.678

In [18]:
df_survivedold=df_survived[df_survived['age']>=60]

In [19]:
p_old_survived=len(df_survivedold)/len(df_survived)
p_old_survived

0.024

In [20]:
p_wo=len(df_women)/len(df)
p_wo

0.3559969442322384

In [21]:
p_we=len(df_wealthy)/len(df)
p_we

0.24675324675324675

In [22]:
p_o=len(df_old)/len(df)
p_o

0.030557677616501147

In [25]:
p_l=len(df_survived)/len(df)
p_l

0.3819709702062643

In [26]:
(p_wealthy_survived * p_women_survived * p_old_survived * p_l)/(p_we * p_wo * p_o)

0.9261936390332054

***Probability of not survival for male passengers from the third class, between the age of 16 and 35

In [27]:
df_died = df[df['survived']==0]
df_men = df[df['sex']=='male']
df_poor = df[df['pclass']==3]

In [28]:
df_young = df[(df['age']>=16) & (df['age']<=35)]

In [29]:
p_men=len(df_men)/len(df)
p_men

0.6440030557677616

In [30]:
p_died=len(df_died)/len(df)
p_died

0.6180290297937356

In [31]:
p_young=len(df_young)/len(df)
p_young

0.46524064171123

In [38]:
p_poor=len(df_poor)/len(df)

In [32]:
df_diedmen=df_died[df_died['sex']=='male']
df_diedpoor=df_died[df_died['pclass']==3]
df_diedyoung=df_died[(df_died['age']>=16) & (df['age']<=35)]

C:\ProgramData\Anaconda2\lib\site-packages\ipykernel_launcher.py:3: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  This is separate from the ipykernel package so we can avoid doing imports until


In [33]:
p_died=len(df_died)/len(df)
p_died

0.6180290297937356

In [35]:
p_men_died=len(df_diedmen)/len(df_died)
p_men_died

0.8430160692212608

In [36]:
p_poor_died=len(df_diedpoor)/len(df_died)
p_poor_died

0.6526576019777504

In [37]:
p_young_died=len(df_diedyoung)/len(df_died)
p_young_died

0.4635352286773795

In [39]:
(p_young_died * p_poor_died * p_men_died * p_died)/(p_men * p_young * p_poor)

0.9712716496751069

***Homework: Probability of not survival for passengers from the third class between the ages of 25 to 35

In [2]:
df_died = df[df['survived']==0]
df_poor = df[df['pclass']==3]
df_25_35 = df[(df['age']>=25) & (df['age']<=35)]

In [3]:
p_died=len(df_died)/len(df)
p_died

0.6180290297937356

In [4]:
p_poor=len(df_poor)/len(df)
p_poor

0.5416348357524828

In [5]:
p_25_35=len(df_25_35)/len(df)
p_25_35

0.24064171122994651

In [6]:
df_diedpoor=df_died[df_died['pclass']==3]
df_died_25_35=df_died[(df_died['age']>=25) & (df['age']<=35)]

C:\Users\Antonina\Anaconda2\lib\site-packages\ipykernel_launcher.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  


In [7]:
p_poor_died=len(df_diedpoor)/len(df_died)
p_poor_died

0.6526576019777504

In [8]:
p_25_35_died=len(df_died_25_35)/len(df_died)
p_25_35_died

0.2373300370828183

In [9]:
(p_25_35_died * p_poor_died * p_died)/(p_25_35 * p_poor)

0.7344622642660756

***Passengers from the third class between the ages of 25 and 35 had 73% chance of not surviving the Titanic disaster